tests/UserRepository-test.js

```diff
@@ -0,0 +1,117 @@
+const chai = require('chai');
+const expect = chai.expect;
+const UserRepository = require('../src/UserRepository');
+const User = require('../src/User');
+
+describe('UserRespository', function() {
+  let userRepository, user1;
+
+  beforeEach(function() {
+    userRepository = new UserRepository([
+      {
+        "id": 1,
+        "name": "Luisa Hane",
+        "address": "15195 Nakia Tunnel, Erdmanport VA 19901-1697",
+        "email": "Diana.Hayes1@hotmail.com",
+        "strideLength": 4.3,
+        "dailyStepGoal": 10000,
+        "friends": [
+          16,
+          4,
+          8
+        ]
+      },
+      {
+        "id": 2,
+        "name": "Jarvis Considine",
+        "address": "30086 Kathryn Port, Ciceroland NE 07273",
+        "email": "Dimitri.Bechtelar11@gmail.com",
+        "strideLength": 4.5,
+        "dailyStepGoal": 5000,
+        "friends": [
+          9,
+          18,
+          24,
+          19
+        ]
+      },
+      {
+        "id": 3,
+        "name": "Herminia Witting",
+        "address": "85823 Bosco Fork, East Oscarstad MI 85126-5660",
+        "email": "Elwin.Tromp@yahoo.com",
+        "strideLength": 4.4,
+        "dailyStepGoal": 5000,
+        "friends": [
+          19,
+          11,
+          42,
+          33
+        ]
+      },
+      {
+        "id": 4,
+        "name": "Mae Connelly",
+        "address": "28926 Schinner Islands, Turnermouth NE 23720-3230",
+        "email": "Marcos_Pollich@hotmail.com",
+        "strideLength": 3.1,
+        "dailyStepGoal": 4000,
+        "friends": [
+          48,
+          7,
+          44,
+          8
+        ]
+      },
+      {
+        "id": 5,
+        "name": "Erick Schaden",
+        "address": "514 Mayert Walk, Jordaneside SC 55023-6523",
+        "email": "Vanessa_Gerhold@gmail.com",
+        "strideLength": 3.1,
+        "dailyStepGoal": 8000,
+        "friends": [
+          13,
+          44,
+          49,
+          33,
+          10
+        ]
+      }]);
+    user1 = new User({
+        "id": 1,
+        "name": "Luisa Hane",
+        "address": "15195 Nakia Tunnel, Erdmanport VA 19901-1697",
+        "email": "Diana.Hayes1@hotmail.com",
+        "strideLength": 4.3,
+        "dailyStepGoal": 10000,
+        "friends": [
+          16,
+          4,
+          8
+        ]
+      });
+  });
+
+  it('should be a function', function() {
+    expect(UserRepository).to.be.a('function');
+  });
+
+  it('should be an instance of UserRepository', function() {
+    expect(userRepository).to.be.an.instanceof(UserRepository);
+  });
+
+  it('should hold all User objects', function() {
+    expect(userRepository.users.length).to.equal(5);
+    expect(userRepository.users[0]).to.be.an.instanceof(User);
+  });
+
+  it('should return a user data given an id', function() {
+    expect(userRepository.getUserData(1)).to.deep.equal(user1)
+  });
```

Comment:
This test could be a bit more clear with referencing the user id and matching it to user data.  It also might be helpful to have a couple of instances here, as well?

Reply:
I refactored this to be more clear about the argument being passed in, the call, and matched it to another instance too!